controllers/hetznerbaremetalhost_controller.go

```diff
@@ -87,6 +90,62 @@ func (r *HetznerBareMetalHostReconciler) Reconcile(ctx context.Context, req ctrl
 		return reconcile.Result{}, err
 	}
 
+	// ----------------------------------------------------------------
+	// Start: avoid conflict errors. Wait until local cache is up-to-date
+	// Won't be needed once this was implemented:
+	// https://github.com/kubernetes-sigs/controller-runtime/issues/3320
+	initialHost := bmHost.DeepCopy()
+	defer func() {
+		if cmp.Equal(initialHost, bmHost) {
```

Comment:
IMO we should check this (and in the other PRs where you added it) in more detail. In which cases do we want to avoid this problem? When is it necessary? I'm sure it won't be necessary in all cases.

Reply:
@janiskemper I see no benefit of reading stale data from the cache. Waiting ~5ms is fine. Error messages which appear again and again because there was a conflict are confusing. What is your goal?